harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -44,22 +44,20 @@ def __init__(self, trial_inst: det.Trial, *args: Any, **kwargs: Any) -> None:
             if util.is_overridden(callback.on_checkpoint_end, pytorch.PyTorchCallback):
                 warnings.warn(
                     "The on_checkpoint_end callback is deprecated, please use "
-                    "on_checkpoint_write_end instead",
+                    "on_checkpoint_write_end instead.",
                     FutureWarning,
                 )
 
-        check.gt_eq(
-            len(self.context.models),
-            1,
-            "Must have at least one model. "
-            "This might be caused by not wrapping your model with wrap_model()",
-        )
-        check.gt_eq(
-            len(self.context.optimizers),
-            1,
-            "Must have at least one optimizer. "
-            "This might be caused by not wrapping your optimizer with wrap_optimizer()",
-        )
+        if not self.context.models:
```

Comment:
Can we avoid truthy checks as substitutions for length checks?

They're valid sometimes, but they're not valid in all cases and it can bite you.  Better to be explicit I think.

Reply:
Oops, fixed.

Reply:
I sometimes slip into using these too.  They look so sleek.  And 95% of the time it's fine, but about 5% of the time it causes issues, and I'm not very good at picking which ones are going to cause issues.